lib/membrane_rtc_engine/endpoints/webrtc/media_event.ex

```diff
@@ -288,6 +280,52 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.MediaEvent do
     end
   end
 
+  defp decode_custom_media_event(%{"type" => "sdpOffer"} = event) do
+    case event do
+      %{
+        "type" => "sdpOffer",
+        "data" =>
+          %{
+            "sdpOffer" => %{
+              "type" => "offer",
+              "sdp" => sdp
+            },
+            "trackIdToTrackMetadata" => track_id_to_track_metadata,
+            "midToTrackId" => mid_to_track_id
+          } = data
+      } ->
+        default_bitrates =
+          Map.new(track_id_to_track_metadata, fn {id, _metadata} -> {id, %{}} end)
+
+        rid_bitrates = Map.get(data, "trackIdToTrackBitrates", %{})
+
+        if is_map(rid_bitrates) do
+          bitrates =
+            Map.new(rid_bitrates, fn {id, bitrate} -> {id, to_track_variants(bitrate)} end)
+
+          {:ok,
+           %{
+             type: :sdp_offer,
+             data: %{
+               sdp_offer: %{
+                 type: :offer,
+                 sdp: sdp
+               },
+               track_id_to_track_metadata: track_id_to_track_metadata,
+               # use default values in VariantSelector if "trackIdToTrackBitrates" is not present
+               track_id_to_track_bitrates: Map.merge(default_bitrates, bitrates),
```

Comment:
Because of the merge, the server could in theory accept an invalid media event, with non exhaustive `track_id_to_track_bitrates` mapping